src/newgrf_debug_gui.cpp

```diff
@@ -402,8 +402,7 @@ struct NewGRFInspectWindow : Window {
 		vseprintf(buf, lastof(buf), format, va);
 		va_end(va);
 
-		offset -= this->vscroll->GetPosition();
-		if (offset < 0 || offset >= this->vscroll->GetCapacity()) return;
+		if (offset + this->vscroll->GetPosition() >= this->vscroll->GetCapacity()) return;
 
 		::DrawString(r.left + LEFT_OFFSET, r.right - RIGHT_OFFSET, r.top + TOP_OFFSET + (offset * this->resize.step_height), buf, TC_BLACK);
```

Comment:
hmm `offset` passed to `DrawString()` is off by `this->vscroll->GetPosition()` I think